src/main/java/com/artipie/Pie.java

```diff
@@ -101,54 +91,7 @@ public Response response(final String line, final Iterable<Map.Entry<String, Str
                     storage -> storage.value(new Key.From(String.format("%s.yaml", repo)))
                 )
                 .thenApply(content -> new RepoConfig(this.vertx, content))
-                .thenCompose(Pie::sliceForConfig)
+                .thenCompose(cfg -> CompletableFuture.completedStage(new SliceFromConfig(cfg)))
```

Comment:
@HDouss `SliceFromConfig` should build `CompletionStage<Slice>` in order to be async and we should remove `completedStage` here

Reply:
@Vatavuk That what the description tell, anyway the constructor is too fast, it is even faster than the construction of the `CompletionStage<Slice>` that was made by `Pie::sliceForConfig`. The actual logic is defered in the `response` medthod.

Reply:
@HDouss If you make `SliceFromConfig` async we don't need `completedStage`. If you think that `SliceFromConfig` doesn't need to be async please say why, because this wasn't the original behaviour.

Reply:
@HDouss I know that the issue says that we should implement `Slice` just to make code more reusable, a refactoring. But I don't see how can we do this unless we change the behaviour.

Reply:
@g4s8 maybe little help here. The purpose of this feature is to break the class into the smaller chunks. It was written with refactoring on the mind and I believe it should not change any behaviour. That being said this PR instantiates `Slice` objects synchronously as opposed to before. Is this ok or should we revert?

Reply:
@Vatavuk Please see my replies earlier. It is true that it is a slice, and not a `CompletionStage<Slice>`, but still what is really synchronous is merely the call to the constructor. ~~And anyway, how to "make SliceFromConfig async" without making it a utility class?~~